permiso/src/main/java/com/greysonparrelli/permiso/PermisoDialogFragment.java

```diff
@@ -48,16 +55,102 @@ public static PermisoDialogFragment newInstance(@Nullable String title, @NonNull
         return dialogFragment;
     }
 
+    public static class Builder {
+        private int titleId = 0;
+        private String title = null;
+        private int buttonTextId = 0;
+        private String message = null;
+        private int messageId = 0;
+        // by default, interpret the dialog msg body as string text
+        private boolean interpretHtml = false;
+        private String buttonText = null;
+
+
+        private Builder() {}
+
+        public Builder(int titleId, int msgBody, int buttonTextId) {
+            this();
+            this.titleId = titleId;
+            this.messageId = msgBody;
+            this.buttonTextId = buttonTextId;
+        }
+
+
+        public int getTitleId() { return titleId; }
+        public String getTitle() { return title; }
```

Comment:
Hmm, I think I see what you're saying. Previously, when the DialogFragment required strings, you needed a context to get the strings from your resIds. That kinds of screwed up your requirements.

I don't know your specific circumstances, but I do know that Permiso will always have a context, since it holds an activity. An interesting thing we do is to require ```Builder.build()``` to take in a Context, which ```Permiso.showRationaleInDialog()``` could provide when it calls ```Builder.build()```.

Then, in the ```Builder.build(Context)``` method, the builder could convert all necessary resources to strings, allowing us to have simple getters like ```getTitle()``` without any other requirements.

What do you think of that?